node/index.ts

```diff
@@ -11,7 +11,19 @@ const SC = bindings(
   'libsignal_client_' + os.platform() + '_' + process.arch
 ) as typeof SignalClient;
 
-export const { initLogger, LogLevel, CiphertextMessageType } = SC;
+export const { initLogger, LogLevel } = SC;
+
+export const enum CiphertextMessageType {
```

Comment:
Why move this out here? (especially since it's still in SC, but that's probably an oversight)

Reply:
Was a hack. Initially tried

export const { CiphertextMessageType } = SC;

to re-export from SC but that gives

```
node/index.ts:1248:11 - error TS2749: 'CiphertextMessageType' refers to a value, but is being used as a type here.

1248   type(): CiphertextMessageType {
               ~~~~~~~~~~~~~~~~~~~~~

```

Reply:
But anyway we don't use CiphertextMessageType within the SC layer, just return the integer value, so IMO best move is just remove it from SC. Thoughts?

Reply:
[Const enums](https://www.typescriptlang.org/docs/handbook/enums.html#const-enums) aren't values, it's true; maybe it'd be `export type CiphertextMessageType = SC.CiphertextMessageType` or something. But at the same time, making this strongly typed is a bit funny regardless, since as you say we mostly care about the integer value, which could be unknown. (The main reason I had it in the .d.ts was so that it could be declared next to the Rust enum with the same constants.)

I'm inclined to say *this* method should *also* return `number` rather than CiphertextMessageType, and not throw on unknown values. That would sidestep the problem too, as long as we can expose the named constants properly.

Reply:
OK changed to return number here, and the app can still compare against the enum values to figure out what they are looking at